judge/pdf_problems.py

```diff
@@ -259,8 +269,55 @@ def _make(self, debug):
         self.log = self.proc.communicate()[0]
 
 
+class SeleniumPDFRender(BasePdfMaker):
+    success = False
+    template = {
+        'printBackground': True,
+        'displayHeaderFooter': True,
+        'headerTemplate': '<div></div>',
+        'footerTemplate': '<center style="margin: 0 auto; font-family: Segoe UI; font-size: 10px">' +
+                          gettext('Page [page] of [topage]').replace('[page]', '<span class="pageNumber"></span>')
+                                                            .replace('[topage]', '<span class="totalPages"></span>') +
+                          '</center>',
+    }
+
+    def _make(self, debug):
+        options = webdriver.ChromeOptions()
+        options.add_argument("--headless")
+        options.add_argument("--disable-gpu")
+        options.binary_location = settings.SELENIUM_CUSTOM_CHROME_PATH
+
+        browser = webdriver.Chrome(settings.SELENIUM_CHROMEDRIVER_PATH, options=options)
+        browser.get("file://" + os.path.abspath(os.path.join(self.dir, 'input.html')))
+        self.log = browser.get_log('driver')
```

Comment:
Is there a need to store the list version in `self.log`? I think the list should be a local variable if possible to avoid `self.log` being a type union. While types don't matter now, I think it's only time before we introduce mypy.

Reply:
Fixed. `self.log` now only contains the string version, however I'm not sure if that is the case for the other pdf generator/renderer logs.